proof/courier.go

```diff
@@ -581,89 +752,50 @@ func (h *HashMailCourier) DeliverProof(ctx context.Context,
 	senderStreamID := deriveSenderStreamID(h.recipient)
 	receiverStreamID := deriveReceiverStreamID(h.recipient)
 
-	// Query delivery log to ensure a sensible rate of delivery attempts.
-	timestamps, err := h.deliveryLog.QueryProofDeliveryLog(
-		ctx, proof.Locator,
-	)
-	if err != nil {
-		return fmt.Errorf("unable to retrieve proof delivery "+
-			"logs: %w", err)
-	}
+	// Interact with the hashmail service using a backoff procedure to
+	// ensure that we don't overwhelm the service with delivery attempts.
+	deliveryExec := func() error {
+		err := h.initMailboxes(
+			ctx, senderStreamID, receiverStreamID,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to initialize "+
+				"mailboxes: %w", err)
+		}
 
-	// Determine whether the historical receiver proof delivery attempts
-	// occurred far enough in the past to warrant a new set of delivery
-	// attempts. Otherwise, wait.
-	//
-	// Only wait if we have a non-zero number of past delivery attempts.
-	timeSinceLastAttempt := timeSinceLastDeliveryAttempt(timestamps)
-	backoffResetWait := h.cfg.BackoffCfg.BackoffResetWait
-	if len(timestamps) > 0 &&
-		timeSinceLastAttempt < backoffResetWait {
+		// Now that the stream has been initialized, we'll write
+		// the proof over the stream.
+		//
+		// TODO(roasbeef): do ecies here
```

Comment:
What could the TODO here mean? Maybe a typo of "retries"? Then I guess it would be addressed now.

Reply:
Not addressed, related to encrypting proofs for the receiver before uploading to the courier:

https://github.com/lightninglabs/taproot-assets/issues/478

Reply:
I've added an additional comment to this TODO